src/objective_types/constraints.jl

```diff
@@ -177,7 +177,7 @@ function TwiceDifferentiableConstraints(c!, lx::AbstractVector, ux::AbstractVect
         # TODO: get rid of this allocation with DI.Cache
         ccache_righttype = zeros(promote_type(T, eltype(_x)), nc)
         c!(ccache_righttype, _x)
-        return sum(_λ[i] * ccache[i] for i in eachindex(_λ, ccache))
+        return LinearAlgebra.dot(_λ, ccache_righttype)
```

Comment:
@pkofod this seems to be a bug that could explain spurious `IPNewton` test failures.

Reply:
good catch